CMake/ThreadMPI.cmake

```diff
@@ -0,0 +1,205 @@
+# notice, this list of conditions and the following disclaimer in the
+# documentation and/or other materials provided with the distribution.
+# 3) Neither the name of the copyright holders nor the
+# names of its contributors may be used to endorse or promote products
+# derived from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY US ''AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL WE BE LIABLE FOR ANY
+# DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+# ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+# If you want to redistribute modifications, please consider that
+# scientific software is very special. Version control is crucial -
+# bugs must be traceable. We will be happy to consider code for
+# inclusion in the official distribution, but derived work should not
+# be called official thread_mpi. Details are found in the README & COPYING
+# files.
+
+include(CheckIncludeFileCXX)
+include(CheckCXXSymbolExists)
+include(CheckCXXSourceCompiles)
+
+# sets TMPI_ATOMICS to 1 if atomic operations are found, unset otherwise
+# Options:
+# include directory for thread_mpi/atomic.h
+MACRO(TMPI_TEST_ATOMICS INCDIR)
+
+    if (NOT DEFINED TMPI_ATOMICS)
+        try_compile(TEST_ATOMICS "${CMAKE_BINARY_DIR}"
+                "${CMAKE_SOURCE_DIR}/CMake/TestAtomics.cpp"
+                COMPILE_DEFINITIONS "-I${INCDIR} -DTMPI_ATOMICS")
+        if (TEST_ATOMICS)
+            message(STATUS "Atomic operations found")
+            # If the check fails, we want to be able to check again,
+            # in case the user has been able to fix this without
+            # needing to delete the cache. Thus we only cache
+            # positive results.
+            set(TMPI_ATOMICS ${TEST_ATOMICS} CACHE INTERNAL "Whether atomic operations are found")
+            set(TMPI_ATOMICS_INCDIR ${INCDIR} CACHE INTERNAL "Atomic operations check include dir")
+        else ()
+            message(STATUS "Atomic operations not found")
+            unset(TEST_ATOMICS)
+        endif()
+    endif()
+
+ENDMACRO(TMPI_TEST_ATOMICS VARIABLE)
+
+try_compile(HAVE_PROCESSOR_NUMBER ${CMAKE_BINARY_DIR} "${CMAKE_SOURCE_DIR}/CMake/TestWinProcNum.cpp")
+
+include(FindThreads)
+
+if(CMAKE_USE_WIN32_THREADS_INIT AND NOT HAVE_PROCESSOR_NUMBER)
+    message(WARNING "Incomplete Windows Processor Group API. If you want GROMACS to be able to set thread affinity, choose a Mingw distribution with a complete API (e.g. Mingw-w64).")
```

Comment:
Fix the message

Reply:
message(WARNING "Incomplete Windows Processor Group API. If you want GOMC to be able to set thread affinity, choose a Mingw distribution with a complete API (e.g. Mingw-w64).")